### ISSUE TITLE
Checking "types" condition in first should also consider nested conditions

### ISSUE THREAD

**Issue by bluwy** [bluwy/publint#47]
```json
"exports": {
  "require": {
    "types": "..."
  },
  "types": "..."
}
```

This is a valid configuration, but `publint` sees `require` before `types` shallowly and naively that it reports a false error.

https://publint.dev/@hebilicious/form-actions-nuxt@0.0.9

https://twitter.com/its_hebilicious/status/1674316686269497348

**Comment by Hebilicious** [comment]
Thanks for the fix 👍🏽
